pkg/sql/catalog/tabledesc/ttl.go

```diff
@@ -24,10 +24,10 @@ func ValidateRowLevelTTL(ttl *catpb.RowLevelTTL) error {
 	if ttl == nil {
 		return nil
 	}
-	if ttl.DurationExpr == "" {
+	if (ttl.DurationExpr == "") == (ttl.ExpirationExpr == "") {
```

Comment:
whilst i like the check simplification, i think it is worth being more explicit:

```
durationisSet := ttl.DurationExpr == ""
expirationExprIsSet = ttl.ExpirationExpr == ""
if durationisSet ||  expirationExprIsSet ...
```

i do wonder whether having `DurationExprIsSet` or `ExpirationExprIsSet` is a good method to have on the `ttl` object (makes checks below easier to read).

Reply:
ping on this comment

Reply:
Added `HasDurationExpr()` and `HasExpirationExpr()` as convenience methods.  TTL requires at least one to be set, but they can be set together.